iotileship/iotile/ship/resources/hardware_manager.py

```diff
@@ -10,7 +10,7 @@
 RESOURCE_ARG_SCHEMA = DictionaryVerifier(desc="hardware_manager arguments")
 RESOURCE_ARG_SCHEMA.add_optional("port", StringVerifier("the port string to use to connect to devices"))
 RESOURCE_ARG_SCHEMA.add_optional("connect", OptionsVerifier(StringVerifier(), IntVerifier(), desc="the uuid of the device to connect to"))
-
+RESOURCE_ARG_SCHEMA.add_optional("connect_direct", OptionsVerifier(StringVerifier(), IntVerifier(), desc="the connection string of the device to connect to"))
```

Comment:
I think connect_direct is always a string.  Even when the value is a single integer, the type of `connection_string` is always a string unless you know of a bug where that isn't the case.

Reply:
that is correct, it should only be a string when connect_direct args is used.